skyportal/onboarding.py

```diff
@@ -124,16 +122,21 @@ def setup_invited_user_permissions(strategy, uid, details, user, *args, **kwargs
         raise Exception("Invitation has already been used.")
 
     group_ids = [g.id for g in invitation.groups]
+    stream_ids = [stream.id for stream in invitation.streams]
+
+    if not all(
+        [
+            stream in invitation.streams
+            for group in invitation.groups
+            for stream in group.streams
+        ]
+    ):
+        raise Exception(
+            "User has not been granted sufficient stream access to be added "
+            "to specified groups."
+        )
 
-    streams = (
-        DBSession()
-        .query(Stream)
-        .join(GroupStream)
-        .filter(GroupStream.group_id.in_(group_ids))
-        .all()
-    )
-    stream_ids = [stream.id for stream in streams]
-
+    # Add user to specified streams
     for stream_id in stream_ids:
```

Comment:
Wait, so it should have been happening as intended, no? (i.e. add stream access first, then add to groups) I wonder why it was misbehaving?

Reply:
@dmitryduev I needed to update where the streams list was coming from (now it comes from invitation.streams, whereas before it was coming from the set of streams associated with the specified groups)